teps/0063-workspace-dependencies.md

```diff
@@ -244,7 +259,255 @@ be handled, or user scenarios that will be affected and must be accomodated.
+
+```go
+// WorkspacePipelineTaskBinding describes how a workspace passed into the pipeline should be
+// mapped to a task's declared workspace.
+type WorkspacePipelineTaskBinding struct {
+	// Name is the name of the workspace as declared by the task.
+	Name string `json:"name"`
+	// Workspace is the name of the workspace declared by the pipeline.
+	Workspace string `json:"workspace"`
+	// SubPath is optionally a directory on the volume which should be used
+	// for this binding (i.e. the volume will be mounted at this sub directory).
+	// +optional
+	SubPath string `json:"subPath,omitempty"`
+	// UseAfter is the list of Task names that should be executed before this Task executes.
+	// UseAfter will be used to construct the Pipeline Directed Acyclic Graph to enforce execution order.
+	// +optional
+	UseAfter []string `json:"useAfter,omitempty"` 
+}
+```
+
+Which will be used as such:
+```yaml
+tasks:
+    - name: task1
+      taskRef:
+        name: write-to-workspace
+      workspaces:
+        - name: output
+          workspace: shared-data
+    - name: task2
+      taskRef:
+        name: read-from-workspace
+      workspaces:
+        - name: input
+          workspace: shared-data
+          useAfter: 
+          - task1
+```
+
+Thus, a `Pipeline` to solve for the [use case](#use-cases) described above would be designed as such:
+
+```yaml
+tasks:
+...
+- name: manual-approval
+  runAfter:
+    - integration-tests
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  workspaces:
+    - name: workarea
+      workspace: pvc
+  taskRef:
+    name: manual-approval
+
+- name: slack-msg
+  workspaces:
+    - name: workarea
+      workspace: pvc
+      useAfter:
+        - manual-approval    
+  taskRef:
+    name: slack-msg
+
+- name: build-image
+  runAfter:
+    - manual-approval
+  taskRef:
+    name: build-image
+
+- name: deploy-image
+  runAfter:
+    - build-image
+  taskRef:
+    name: deploy-image
+```
+
+This proposal provides the simplest approach to solve the problem of `Workspace` dependencies. Moreover, `useAfter` 
+is parallel to `runAfter`, making it familiar to users and improving its learnability. 
+
+## Design Evaluation
+
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+#### Reusability
+
+By enabling users to specify resource dependencies based on `Workspaces`, we prevent unexpected failures when we 
+roll out skipping and failure strategies for common CI/CD use cases. Without this, users would have needed to build 
+workarounds in their `Pipelines` to address those failures thus making them less reusable. Hence, this proposal 
+maintains the reusability of Tekton `Pipelines`.
```

Comment:
Could we show/describe what those workarounds are and how they reduce the re-usability ?